command/commands/team.py

```diff
@@ -476,3 +486,67 @@ def delete_helper(self, team_name, user_id) -> ResponseTuple:
             logging.error("team delete unsuccessful")
             return f"Team delete was unsuccessful with " \
                    f"the following error: {e.data}", 200
+
+    def refresh_helper(self, user_id) -> ResponseTuple:
+        """
+        Ensure that the local team database is the same as GitHub's.
+
+        In the event that our local team database is outdated compared to
+        the teams on GitHub, this command can be called to fix things.
+        :return: return error message if user has insufficient permission level
+                 otherwise returns success messages with # of teams changed
+        """
+        num_changed = 0
+        num_added = 0
+        num_deleted = 0
+        modified = []
+        try:
+            command_user = self.facade.retrieve(User, user_id)
+            if not check_permissions(command_user, None):
+                return self.permission_error, 200
+            local_teams: List[Team] = self.facade.query(Team)
+            remote_teams: List[GithubTeam.Team] = self.gh.org_get_teams()
+            local_ids = dict((team.github_team_id, team)
+                             for team in local_teams)
+            remote_ids = dict((str(team.id),
+                               Team(str(team.id), team.name, ""))
```

Comment:
As far as I'm aware, this does not populate the `members` attribute. You must do so, maybe with a small helper function. Some kind of utility function `convert_GithubTeam_to_DBTeam` or something shorter.

Because, later, you will find that there are no members in the Team (meaning that it is modified), and you will be replacing the set of members (probably quite accurate) with an empty set, effectively Thanos-snapping them out of existence.

Please nerf.

Reply:
ahk, I knew not having teams containing members was going to bite me. i will get on it promptly, and by promptly i mean sometime between now and the heat death of the universe

Reply:
Very accurate; just not very precise.:)

Reply:
should I just make the github interface's get_teams return actual teams we can work with, rather than github ones?

Reply:
yes